app/models/test_passage.rb

```diff
@@ -52,4 +61,16 @@ def correct_answer?(answer_ids)
   def correct_answers
     current_question.answers.correct
   end
+
+  def timer
+    test.timer * 60
+  end
+
+  def time_passed
+    Time.zone.now - created_at
+  end
+
+  def end_test
+    self.current_question = test.questions.last
```

Comment:
здесь скорее подойдет
```suggestion
    self.current_question = nil
```
у нас же именно так завершенность теста опеределяется
посмотри на метод `completed?`
в этом случае в контроллере будет редирект на страницу завершения теста сразу, что нам и нужно